sdk/python/feast/infra/offline_stores/third_party/spark_source.py

```diff
@@ -0,0 +1,231 @@
+import pickle
+from typing import Any, Callable, Dict, Iterable, Optional, Tuple
+
+from feast_spark_offline_store.spark_type_map import spark_to_feast_value_type
+from pyspark.sql.utils import AnalysisException
+
+from feast.data_source import DataSource
+from feast.errors import DataSourceNotFoundException
+from feast.protos.feast.core.DataSource_pb2 import DataSource as DataSourceProto
+from feast.protos.feast.core.SavedDataset_pb2 import (
+    SavedDatasetStorage as SavedDatasetStorageProto,
+)
+from feast.repo_config import RepoConfig
+from feast.saved_dataset import SavedDatasetStorage
+from feast.value_type import ValueType
+
+
+class SparkSource(DataSource):
+    def __init__(
+        self,
+        table: Optional[str] = None,
+        query: Optional[str] = None,
+        # TODO support file readers
+        # path: Optional[str] = None,
+        # jdbc=None,
+        # format: Optional[str] = None,
+        # options: Optional[Dict[str, Any]] = None,
+        event_timestamp_column: Optional[str] = None,
```

Comment:
feast today does some event timestamp inference in https://github.com/feast-dev/feast/blob/master/sdk/python/feast/inference.py#L99-L99

Do we need to support this somehow too? it basically checks if there is only one timestamp column